### PR TITLE
Change Create Recipient CLI to use global metastore ID

### PR DESCRIPTION
**Pull request by wchau**
Sharing code is not used anymore.

Tested locally.

### CODE REVIEW

**Comment by wchau** [comment]
Do we still need to fix public API even if sharing code is useless today?

**Review by adamcain-db — COMMENTED** [review]
The databricks-cli changes LGTM, though I know nothing of the underlying CreateRecipient API change. Should a delta-sharing person also review?

**Review by zhuansunxt — COMMENTED** [review]
In the UI we use the term `sharing identifier`. I think the rationale behind that is we want to be more abstract so we can change the format later. Let's keep it consistent in the CLI. (probably too late to migrate REST API)

File: databricks_cli/unity_catalog/delta_sharing_cli.py [hunk 1]
```diff
@@ -183,8 +183,8 @@ def delete_share_cli(api_client, name):
 @click.option('--name', required=True, help='Name of new recipient.')
 @click.option('--comment', default=None, required=False,
               help='Free-form text description.')
-@click.option('--global-metastore-id', default=None, required=False,
-              help='The global UC metastore id provided by the data recipient offline.')
+@click.option('--sharing-id', default=None, required=False,
+              help='The sharing id provided by the data recipient offline.')
```
Comment (zhuansunxt) [inline databricks_cli/unity_catalog/delta_sharing_cli.py:187]: nit: "The sharing identifier ...", to be exactly consistent as UI.

**Review by zhuansunxt — APPROVED** [review]
Thanks for the fix!

**Review by pietern — APPROVED** [review]
We can ignore the API compatibility test as these functions are unlikely to already be in use.

**Pull request merged**